Add validation to MergedLFNBase/UnmergedLFNBase arguments

We have these LFNs defined here:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ReqMgr/Tools/cms.py#L68

and unmerged just below it. I suggest we create a validate function in the Lexicon module and call that function from the StdSpecs validate function.

Or even better, we could have these MergedLFNBase lfns in the aux db

In addition to that:
* remove the assignment default values
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMSpec/StdSpecs/StdBase.py#L1092
* I don't know how correct are those "/store/results/" lfns, thus I'd suggest to validate it on `startswith('/store/results/')` and that's all, they are rarely used anyways and we can always enforce it to be assigned via script.

@ticoann what do you think?
BTW, slightly related to #7979

The current validation breaks the Tier0.

AssertionError: '/store/backfill/1/data/Tier0_REPLAY_vocms001/HLTPhysics/RAW/v125' does not match regular expression /store/results/([a-zA-Z0-9\-_]+)/([a-zA-Z0-9\-_]+)/([a-zA-Z0-9\.\-_]+)/([A-Z\-_]+)/([a-zA-Z0-9\-_]+)

This prevents the latest T0 release from being used (it "worked" before since we didn't override the LFN at assignment and got the wrong/default values).

--- a/src/python/WMCore/WMSpec/StdSpecs/StdBase.py
+++ b/src/python/WMCore/WMSpec/StdSpecs/StdBase.py
@@ -1088,8 +1088,8 @@ class StdBase(object):
                                               "validate": lambda x: x in ["Low", "Normal", "High"]},
                      "DeleteFromSource": {"default": False, "type": strToBool},
                      # merge settings
-                     "UnmergedLFNBase": {"default": "/store/unmerged"},
-                     "MergedLFNBase": {"default": "/store/data"},
+                     "UnmergedLFNBase": {"assign_optional": True},
+                     "MergedLFNBase": {"assign_optional": True},
                      "MinMergeSize": {"default": 2 * 1024 * 1024 * 1024, "type": int,
                                       "validate": lambda x: x > 0},
                      "MaxMergeSize": {"default": 4 * 1024 * 1024 * 1024, "type": int,

--- a/src/python/WMCore/WMSpec/WMWorkload.py
+++ b/src/python/WMCore/WMSpec/WMWorkload.py
@@ -1724,13 +1724,13 @@ class WMWorkloadHelper(PersistencyHelper):
         self.setTrustLocationFlag(inputFlag=strToBool(kwargs["TrustSitelists"]),
                                   pileupFlag=strToBool(kwargs["TrustPUSitelists"]))
 
-        # FIXME not validated
-        self.setLFNBase(kwargs["MergedLFNBase"], kwargs["UnmergedLFNBase"])
-
         self.setMergeParameters(int(kwargs["MinMergeSize"]),
                                 int(kwargs["MaxMergeSize"]),
                                 int(kwargs["MaxMergeEvents"]))
 
+        # FIXME not validated
+        if kwargs.get("MergedLFNBase") and kwargs.get("UnmergedLFNBase"):
+            self.setLFNBase(kwargs["MergedLFNBase"], kwargs["UnmergedLFNBase"])
         # Set ProcessingVersion and AcquisitionEra, which could be json encoded dicts
         # it should be processed once LFNBase are set
         if kwargs.get("AcquisitionEra") is not None:

